connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -525,6 +525,7 @@ private WorkerTask buildWorkerTask(ClusterConfigState configState,
         producerProps.put(ProducerConfig.ACKS_CONFIG, "all");
         producerProps.put(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION, "1");
         producerProps.put(ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG, Integer.toString(Integer.MAX_VALUE));
+        producerProps.put(ProducerConfig.CLIENT_ID_CONFIG, "connector-producer-" + id);
```

Comment:
Coming back to this, it occurred to me that the client IDs will only be unique across all clients of this Kafka cluster if the connector name is unique. That is the case within a single Connect cluster, but it is not guaranteed or checked in any way across multiple clusters.

Do we want to ensure that the client ID is indeed _more_ unique? Or is this sufficient considering that the client ID is more for tracing calls, and duplicates are tolerated?

Reply:
Thanks @rhauch - also just getting back to this. Good point. Still, based on my own experience I think this is an adequate level of uniqueness.  In our KC clusters we automatically apply unique cluster name prefixes to all JMX names as we export them. This covers the KC metrics, as well as JVM metrics (Garbage collector metrics etc) which people typically need to track. Also, as you say, duplicates introduced accidentally at this level can usually be tolerated and corrected with minimal trouble.

Reply:
For MirrorMaker 2, which spins up multiple workers with different configurations but the same Connector names, this poses a problem. This is an edge case which MM2 can work around, but it would be nice if we could solve the problem here.

I'd like the client IDs to look like:

MyConnector-1-producer-2

where MyConnector is the Connector name, and MyConnector-1 is the workerTaskId.